libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/transforms/ClasspathEntrySnapshotTransform.kt

```diff
@@ -40,26 +42,11 @@ abstract class ClasspathEntrySnapshotTransform : TransformAction<ClasspathEntryS
         val classpathEntryInputDirOrJar = inputArtifact.get().asFile
         val snapshotOutputFile = outputs.file(classpathEntryInputDirOrJar.name.replace('.', '_') + "-snapshot.bin")
 
-        // There are two levels of granularity when taking a snapshot:
-        //   - CLASS_LEVEL (coarse-grained): The size of the snapshot is smaller, but we will have coarse-grained classpath changes, which
-        //     means more source files will be recompiled.
-        //   - CLASS_MEMBER_LEVEL (fine-grained): The size of the snapshot is larger, but we will have fine-grained classpath changes, which
-        //     means fewer source files will be recompiled.
-        // Therefore, CLASS_LEVEL is suitable for classes that are infrequently changed (e.g., external libraries which are typically
-        // stored/transformed inside the Gradle user home, plus a few hard-coded cases below), whereas CLASS_MEMBER_LEVEL is suitable for
-        // classes that are frequently changed (e.g., classes produced by the current project).
-        val granularity = if (
-            classpathEntryInputDirOrJar.startsWith(parameters.gradleUserHomeDir.get().asFile) ||
-            classpathEntryInputDirOrJar.name == "android.jar" ||
-            classpathEntryInputDirOrJar.name.startsWith("kotlin-compiler-embeddable")
-        ) {
-            ClassSnapshotGranularity.CLASS_LEVEL
-        } else {
-            ClassSnapshotGranularity.CLASS_MEMBER_LEVEL
-        }
-        val buildMetricsReporterService = parameters.buildMetricsReporterService.orNull
+        val granularity = getClassSnapshotGranularity(classpathEntryInputDirOrJar, parameters.gradleUserHomeDir.get().asFile)
```

Comment:
nit: Get the gradleUserHomeDir directly in the function?

Reply:
This has the benefit of listing all the inputs of the function in the arguments, so it's more modular. But it's not very important, just let me know if you still prefer the other way.